### DISCUSSION TITLE
Make `watch` behave the same way as `run` when executed from a specific package directory

### DISCUSSION THREAD

**Discussion by Ugzuzg** [vercel/turborepo#9317] [Ideas]
### Goals

1. Identical behaviour for `run` and `watch` when run from a specific package directory, with the only difference being that `watch` watches

### Background

Currently, if I `cd packages/a` and run `turbo run build`, it will only use that package's `build` command as a target, only running its dependencies and then the build of that package.
If I `cd packages/a` and run `turbo watch build`, it suddenly looks for all `build` commands in the workspace, and I have to add `--filter` to limit the scope of the watcher.

### Proposal

Make `run` and `watch` the same.

**Comment by anthonyshew** [comment]
Hey, we've now got this done in https://github.com/vercel/turborepo/pull/9404. It will be a part of the next canary release (v2.2.4-canary.9) and later.

Thanks for suggesting!
